File: toolknit-desktop/src-tauri/src/platform/qa-devtools.js
(() => {
  const installed = Symbol.for('toolknit.qa-devtools');
  if (window[installed]) return;
  window[installed] = true;
  let pending = false;
  document.addEventListener('keydown', event => {
    if (event.code !== 'F12' && event.key !== 'F12') return;
    event.preventDefault();
    event.stopImmediatePropagation();
    if (event.repeat || pending) return;
    pending = true;
    window.__TAURI_INTERNALS__.invoke('plugin:webview|internal_toggle_devtools')
      .catch(() => console.error('Could not toggle test-package DevTools.'))
      .finally(() => { pending = false; });
  }, true);
})();

File: toolknit-desktop/src-tauri/src/platform/qa_devtools.rs
pub fn install<R: tauri::Runtime>(webview: &tauri::Webview<R>) {
    if webview.label() != "main" {
        return;
    }
    if let Err(error) = webview.eval(include_str!("qa-devtools.js")) {
        log::warn!("Could not install test-package DevTools shortcut: {error}");
    }
}
